[WIP] Rewrite navbar

dawidpotocki opened the pull request:
Fixes #877, Fixes #872, Closes #1195.

No Javascript needed, that's why I even started it :)

- [x] Dropdowns
- [x] Hide dropdown when clicking anything else
- [x] Mobile Hamburger

![image](https://user-images.githubusercontent.com/38681822/62938226-24c77c80-bdbe-11e9-8ace-9eeb34ab9f65.png)

dawidpotocki commented:
> I don't understand the code, but it seems to work even if it's a bit funny how everything can be opened simultaneously.

```
<details>
	<summary>
		This is summary!
	</summary>
	This is some text
</details>
```

<details>
	<summary>
		This is summary!
	</summary>
	This is some text
</details>


Now, how I made this thing ^ work like a dropdown? I had to first create
a `<span>` after `</summary> and make `position: absolute` so it would
"fly" and not show itself in the navbar, which would destroy everything.
Next to fix bug with "Services" dropdown being cut of, I had to set
`<details>` to `positon: relative` so dropdown is shown next to it and
then `right: 0` to the dropdown so it would go to the left side.

It's pretty simple, considering that one from BS4 is using Javascript
for the same thing and everything you need is builtin to the HTML!

nitrohorse commented on assets/css/style.scss:
```diff
@@ -182,4 +182,38 @@ footer {
+ */
+
+.nav-text {
+  color: #9a9da0;
```
Small suggestion, since we're using SASS could we use color variables for these new classes to allow the colors to be re-usable? I think like this:
```
$color-gray: #9a9da0;

.nav-text {
  color: $color-gray;
}
```

dawidpotocki replied:
I don't think that there is need for that as it probably will not be used anywhere else.
It was the color used before in navbar, that was probably built-in to BS4.
But if you want, I can do that.

nitrohorse replied:
Ah, that makes sense. I suppose we can make the colors re-usable if we need to reference them again in the future somewhere else.

JonahAragon replied:
The actual color is `rgba(255,255,255,0.5)`, FYI

nitrohorse commented:
This is really cool! Though I think we'll need `Hide dropdown when clicking anything else` to be resolved; the usability without isn't great.

dawidpotocki commented:
> Though I think we'll need Hide dropdown when clicking anything else to be resolved; the usability without it isn't great.

Got it working with JS. I don't believe it is possible without it. But at least navbar dropdowns _sort of work_, which was not the case before.

JonahAragon commented on assets/css/style.scss:
```diff
@@ -182,4 +182,41 @@ footer {
+
+.nav-link:hover,
+.nav-summary:hover {
+  color: #c4c9cc;
```
This `:hover` color seemingly does not apply to navbar items without dropdowns (i.e. the home button, participate link, and donate link).

dawidpotocki replied:
It is because I renamed `.nav-link` to `.nav-anchor`, because it was
already defined by Bootstrap, so I renamed it and forgot to update it.
Should be good now.

(if you got the same reply on emails, sorry but GitHub does not like me replying to these comments with email :C)

dawidpotocki commented:
It is because I renamed `.nav-link` to `.nav-anchor`, because it was
already defined by Bootstrap, so I renamed it and forgot to update it.
Should be good now.

dawidpotocki commented:
> > +/*
> + * Navbar
> + */
> +
> +.text-nav {
> +  color: #9a9da0;
> +}
> +
> +.nav-anchor {
> +  display: inline;
> +  color: inherit;
> +}
> +
> +.nav-link:hover,
> +.nav-summary:hover {
> +  color: #c4c9cc;
>
> This `:hover` color seemingly does not apply to navbar items without dropdowns (i.e. the home button, participate link, and donate link).

It is because I renamed `.nav-link` to `.nav-anchor`, because it was
already defined by Bootstrap, so I renamed it and forgot to update it.
Should be good now.

Perelandra0x309 commented:
W3.CSS does nice navbars with no JS.
https://www.w3schools.com/w3css/w3css_navigation.asp

dawidpotocki commented:
> W3.CSS does nice navbars with no JS.

They are on hover, this one is on click. The collapsing one is using
JavaScript, so it's not really useful.

Perelandra0x309 commented:
They do have clickable ones too.  The menu on that website itself is a clickable menu, so can be used as an example.

dawidpotocki commented:
> They do have clickable ones too.  The menu on that website itself is a clickable menu, so can be used as an example.

They. Require. JavaScript.

JonahAragon commented:
This is coming along great! I wonder if you would be able to push the dropdown box down a bit...

Old:

<img width="391" alt="image" src="https://user-images.githubusercontent.com/3637842/63313388-d6d4dc00-c2c9-11e9-827a-957e0fd4b094.png">

Current in PR:

<img width="391" alt="image" src="https://jda.mn/z5rqw">

For some reason that dropdown being so close to the navbar text is unsettling to me. Also, want to make sure changing the color of an active dropdown to white is on your to-do list. I would imagine `summary:focus` would do the trick, maybe not perfectly but good enough, unless you have other plans.

dawidpotocki commented:
> This is coming along great! I wonder if you would be able to push the dropdown box down a bit...
> For some reason that dropdown being so close to the navbar text is unsettling to me. Also, want to make sure changing the color of an active dropdown to white is on your to-do listw I would imagine `summary:focus` would do the trick, maybe not perfectly but good enough, unless you have other plans.

Yeah, I'm planning to do that, but my priority is making a mobile
hamburger ui now. I'm trying to implement it without JavaScript and it
seems to be possible, all the possible ways are weird hacks, but
whatever.

Perelandra0x309 commented:
> They. Require. JavaScript.

Where do you see that?  I am using it just fine without JS.

dawidpotocki commented:
> Where do you see that? I am using it just fine without JS.

https://invidio.us/watch?v=mNttgpgNFS4

Perelandra0x309 commented:
If you insist on having clickable menus (on desktop) then yes JS in needed for that one feature.  But I thought you wanted this to be JS free?  Are hoverable menus that bad?  Those do not require JS.  Also the hoverable menus from W3 are actually clickable on mobile devices.

BTW W3 is just a framework of standard CSS attributes, just makes it easier to do what you want.  You are probably doing the exact same with with your straight CSS if you are able to get hoverable menus without JS.  So it's a decision you and the site owners need to make- use JS to get clickable menus or drop JS and hav menus be hoverable on desktop (clickable on mobiles).

dawidpotocki commented:
> If you insist on having clickable menus (on desktop) then yes JS in needed for that one feature.

But. I. Implemented. Them. In. This. Pull. Request. Without. JavaScript.

Perelandra0x309 commented:
Ah, OK your last update from yesterday indicated you were having to use "weird hacks" to do it without JS.  If you got it working, great!

dawidpotocki commented:
__Everything is finished.__

I think.
